.github/workflows/build-test.yml

```diff
@@ -1171,7 +1171,7 @@ jobs:
     
     # Skipped tasks: benchmark, memory, datasize, binsize
     needs: [check-msrv, test, testdata, test-docs, full-datagen, ffi, verify-ffi, gn, features, wasm, fmt, tidy, clippy, doc, ]
-    if: ${{ always() && contains(needs.*.result, 'failure') && (github.event_name == 'schedule' || github.event_name == 'workflow_dispatch') }}
+    if: ${{ always() && contains(needs.*.result, 'failure') && (github.event_name == 'schedule' || github.event_name == 'workflow_dispatch' || github.event_name == 'push') }}
```

Comment:
issue: for the last one we should `&& branch == main`

Reply:
This workflow only runs on push if the branch is main.

Reply:
Yes, but I would like to have some redundancy on that